bindings/rust/bench/benches/handshake.rs

```diff
@@ -2,77 +2,72 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use bench::{
-    CipherSuite, CryptoConfig,
-    ECGroup::{self, *},
-    HandshakeType::{self, *},
-    OpenSslHarness, RustlsHarness, S2NHarness,
-    SigType::{self, *},
-    TlsBenchHarness,
+    CipherSuite, CryptoConfig, HandshakeType, KXGroup, OpenSslConnection, RustlsConnection,
+    S2NConnection, SigType, TlsConnPair, TlsConnection,
 };
 use criterion::{
     criterion_group, criterion_main, measurement::WallTime, BatchSize, BenchmarkGroup, Criterion,
 };
+use strum::IntoEnumIterator;
 
 pub fn bench_handshake_params(c: &mut Criterion) {
-    fn bench_handshake_for_library<T: TlsBenchHarness>(
+    fn bench_handshake_for_library<T: TlsConnection>(
```

Comment:
Can we move these functions outside of the main `bench_handshake_params` function? It's little confusing to have nested function declarations. Same for the throughput benchmark.